src/Volume.ts

```diff
@@ -244,19 +246,32 @@ export default class Volume {
     let noReload =
```

Comment:
this is just a readability comment: after reading through this, I think I would divide this into a function that handles all the required checks, including the call to get the `dims` and then an update function that just does lines 264- 274

Reply:
Based on your comments and Dan's above, I broke out a bunch of functionality from this function into new ones. The new `updateRequiredData` should be significantly slimmer and more comprehensible.